specs/monitoring_integration_service.adoc

```diff
@@ -0,0 +1,207 @@
+= Spec - Monitoring-integration as a service
+
+== Introduction
+
+Monitoring Integration component is responsible for the integration of tendrl with grafana.
+It performs the following functionalities:
+
+* Uploads default dashboards to grafana.
+
+== Problem description
+
+In the present implementation of monitoring-integration to upload dashboards in
+grafana the user has to run the code manually every time which is not feasible.
+Monitoring integration will be providing other functionalities such as
+
+* Handling alerts generated by grafana
+* Making dynamic configuration changes in graphite and grafana
+
+for which it will be required to be implemented as a service.
+
+== Proposed change
+
+Monitoring-integration component is changed to a service based on the tendrl structure.
+Instead of starting each and every times manually, it is installed and started once
+as a service in the background.
+
+To convert as a service monitoring-integration component needs some additional
+modules on existing structure.
+
+1.Manager
+
+2.Namespace
+
+3.Object
+
+Mananger:
+
+Monitoring-integration execution flow is started from the manager modules. Manager
+module is the entry point for monitoring-integration, all the gevents are initialized
+and started from the manager modules. A manager can kill all the gevent when
+monitoring-integration service stops.
+
+When monitoring-integration start manager module will do the all configuration
+using configuration files. It will also create a separate namespace for the monitoring-integration.
+the namespace will be called "NS.monitoring"
+
+Namespace:
+
+A manager will create its own namespace for the monitoring-integration to separate
+its own objects from the other modules. These objects are only accessible by
+monitoring-integration.
+
+Object:
+
+Each tendrl component has its own set of objects for a different purpose. Each
+object have its own structure. Monitoring-integration have an config and definition
+object.
+
+```
+Sample config class for monitoring-integration:
+
+class Config(objects.BaseObject):
+    internal = True
+
+    def __init__(self, config=None, *args, **kwargs):
+        self._defs = {}
+        super(Config, self).__init__(*args, **kwargs)
+
+        self.data = config or cmn_config.load_config(
+            'node-agent', "/etc/tendrl/monitoring-integration/monitoring-integration.conf.yaml")
+        self.value = '_NS/monitoring/config'
+
+
+Sample definition class for monitoring-integration:
+
+class Definition(objects.BaseObject):
+    internal = True
+
+    def __init__(self, *args, **kwargs):
+        self._defs = {}
+        super(Definition, self).__init__(*args, **kwargs)
+
+        self.data = pkg_resources.resource_string(__name__, "monitoring_integration.yaml")
+        self._parsed_defs = yaml.safe_load(self.data)
+        self.value = '_NS/monitoring/definitions'
+```
+
+Install and run:
+
+Monitoring-integration will be installed in service node only. It will be
```

Comment:
What is a "service node"? There's no such term in the architecture documentation.